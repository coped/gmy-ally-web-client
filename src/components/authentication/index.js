import Login from "components/authentication/Login";
import LoginForm from "components/authentication/LoginForm";
import SignupForm from "components/authentication/SignupForm";

export { Login, LoginForm, SignupForm };
