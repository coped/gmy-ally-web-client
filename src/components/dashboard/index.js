import Dashboard from "components/dashboard/Dashboard";

export { Dashboard };
