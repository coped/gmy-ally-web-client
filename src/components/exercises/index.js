import Exercises from "components/exercises/Exercises";

export { Exercises };
