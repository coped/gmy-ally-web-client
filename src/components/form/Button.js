import React from "react";

export default function Button(props) {
  const { children, classModifiers } = props;
  const isLoading = props.isLoading ? "is-loading" : "";
  return (
    <div className="field">
      <div className="control">
        <button className={`button ${classModifiers} ${isLoading}`}>
          {children}
        </button>
      </div>
    </div>
  );
}
