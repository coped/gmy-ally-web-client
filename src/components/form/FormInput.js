import React from "react";

export default function Input(props) {
  const { label, ...rest } = props;
  return (
    <div className="field">
      <label className="label">{label}</label>
      <div className="control">
        <input className="input" {...rest} />
      </div>
    </div>
  );
}
