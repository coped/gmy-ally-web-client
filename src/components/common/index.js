import Notification from "components/common/Notification";
import Button from "components/common/Button"

export { Notification, Button };
