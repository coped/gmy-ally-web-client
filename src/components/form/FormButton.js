import React from "react";
import { Button } from "components/common";

export default function formButton(props) {
  const { children, ...rest } = props;
  return (
    <div className="field">
      <div className="control">
        <Button {...rest}>{children}</Button>
      </div>
    </div>
  );
}
