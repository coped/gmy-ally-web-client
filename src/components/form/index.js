import FormButton from "components/form/FormButton";
import FormInput from "components/form/FormInput";

export { FormButton, FormInput };
