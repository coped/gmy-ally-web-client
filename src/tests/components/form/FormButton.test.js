import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { FormButton } from "components/form";

Enzyme.configure({ adapter: new Adapter() });

describe("FormButton", () => {
  const props = { id: "this-button", classList: ["is-info"] };
  const message = "I'm a button!";

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<FormButton />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders props without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<FormButton {...props}>{message}</FormButton>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(
      <FormButton {...props}>{message}</FormButton>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
