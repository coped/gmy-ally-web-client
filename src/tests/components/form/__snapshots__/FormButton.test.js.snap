// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButton has a valid snapshot 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <button
      className="button is-info"
      id="this-button"
    >
      I'm a button!
    </button>
  </div>
</div>
`;
